Apps/WebClient/src/ClientApp/src/views/ProfileView.vue

```diff
@@ -302,14 +302,15 @@ export default class ProfileView extends Vue {
     }
 
     private validations(): unknown {
-        const validPhoneNumberFormat = async (rawInputSmsNumber: string) => {
+        const validPhoneNumberFormat = (rawInputSmsNumber: string) => {
+            if (!rawInputSmsNumber) {
+                return true;
+            }
             if (!ValidationRegEx.PhoneNumberMasked.test(rawInputSmsNumber)) {
                 return false;
             }
             const phoneNumber = PhoneUtil.stripPhoneMask(rawInputSmsNumber);
-            return await this.userProfileService.isPhoneNumberValid(
-                phoneNumber
-            );
+            return this.userProfileService.isPhoneNumberValid(phoneNumber);
```

Comment:
I'm curious why the function was made non-async.

Reply:
When returning the await result Vuelidate's `$pending` wasn't working as expected, when I changed it to return the result from the http request which is also a promise it worked. This works out of the box with Vuelidate so I didn't tweak any further.

https://vuelidate.js.org/#sub-asynchronous-validation